### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by vlad12345123**

### CODE REVIEW

**Review by GeranSkot — APPROVED** [review]

**Review by Atalay-Polat — APPROVED** [review]

File: exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs [hunk 1]
```diff
@@ -22,20 +22,26 @@ public async Task<List<Question>> GetByExpressionAsync(Expression<Func<Question,
...
         {
-            Expression<Func<Question, bool>> takeGrammerQuestions = q => q.QuestionType == QuestionType.Grammar;
+            Expression<Func<Question, bool>> takeGrammerQuestions = q => q.QuestionType == QuestionType.Grammar && q.LevelType == levelType;
             return await GetByExpressionAsync(takeGrammerQuestions, 10);
         }
 
         public async Task<List<Question>> GetAuditionQuestionsAsync(LevelType levelType)
         {
-            Expression<Func<Question, bool>> takeAuditionQuestions = q => q.QuestionType == QuestionType.Audition && q.LevelType == levelType;
+            var url = _appDbContext.Questions
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs:39]: Url is object?

File: exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs [hunk 2]
```diff
@@ -22,20 +22,26 @@ public async Task<List<Question>> GetByExpressionAsync(Expression<Func<Question,
...
+                .Select(x => x.FileUrl)
+                .OrderBy(x => Guid.NewGuid())
+                .FirstOrDefault();
+
+            Expression<Func<Question, bool>> takeAuditionQuestions = q => q.QuestionType == QuestionType.Audition && q.LevelType == levelType && q.FileUrl == url;
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs:44]: Object = string?

File: exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs (hunk 1 above)
Comment (vlad12345123) [reply exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs:39]: url is string O_0

File: exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs (hunk 2 above)
Comment (vlad12345123) [reply exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs:44]: string == string O_O

**Review by romanstat — APPROVED** [review]

**Pull request merged**